### PR TITLE
[WIP] Add conflict integration tests & fix an edge case

### PR DESCRIPTION
**Pull request by aenario**

### CODE REVIEW

File: test/integration/conflict_resolution.js [hunk 1]
```diff
@@ -150,6 +150,107 @@ describe('Conflict resolution', () => {
...
+    await helpers.remote.pullChanges()
+    await helpers.syncAll()
+    await helpers.local.scan()
+    await helpers.remote.pullChanges()
+    await helpers.syncAll()
+    should(await helpers.trees()).deepEqual({remote: expected, local: expected})
+  }
+
+  describe('create 2 files same name', () => {
+    it('file local & folder remote', async () => {
```
Comment (taratatach) [inline test/integration/conflict_resolution.js]: Don't we have 2 files in this context?

File: test/integration/conflict_resolution.js [hunk 2]
```diff
@@ -150,6 +150,107 @@ describe('Conflict resolution', () => {
...
   })
 
+  const afterFullSyncBothTreeDeepEqual = async (expected) => {
+    await helpers.local.scan()
+    await helpers.remote.pullChanges()
+    await helpers.syncAll()
+    await helpers.local.scan()
+    await helpers.remote.pullChanges()
+    await helpers.syncAll()
+    should(await helpers.trees()).deepEqual({remote: expected, local: expected})
```
Comment (taratatach) [inline test/integration/conflict_resolution.js]: Maybe the newly created `mergeSideEffects` helper would be useful here?

File: test/integration/conflict_resolution.js [hunk 3]
```diff
@@ -150,6 +150,107 @@ describe('Conflict resolution', () => {
...
+        'same-name-conflict-...',
+        'same-name/'
+      ]
+
+      await afterFullSyncBothTreeDeepEqual(expectedTree)
+      await afterFullSyncBothTreeDeepEqual(expectedTree)
+    })
+  })
+
+  describe('move folder to conflict', () => {
```
Comment (taratatach) [inline test/integration/conflict_resolution.js]: I don't really understand what is meant here.

File: test/support/helpers/context_dir.js [hunk 4]
```diff
@@ -123,7 +123,7 @@ class ContextDir {
   }
 
   async move (src /*: string */, dst /*: string */) {
-    return fs.move(this.abspath(src), this.abspath(dst))
+    return fs.rename(this.abspath(src), this.abspath(dst))
```
Comment (taratatach) [inline test/support/helpers/context_dir.js]: What's the difference between those 2?

**Comment by taratatach** [comment]
We're not just adding an integration test here, we're apparently fixing a bug. I think we should make this clear in the commit & PR titles.

File: test/support/helpers/context_dir.js (hunk 4 above)
Comment (sebn) [reply test/support/helpers/context_dir.js]: [`Local#_move()`](https://github.com/cozy-labs/cozy-desktop/blob/v3.11.1-beta.1/core/local/index.js#L342) actually uses [`fs.rename()`](https://github.com/cozy-labs/cozy-desktop/blob/v3.11.1-beta.1/core/local/index.js#L359), so this better matches our own sync actions.

Also `fs.move` actually comes from `fs-extra`, not `fs`, it does some black magic to ensure move succeeds in weird cases (e.g. separate volumes), and may end up changing the inode of watched files.

File: test/integration/conflict_resolution.js (hunk 2 above)
Comment (sebn) [reply test/integration/conflict_resolution.js]: This is an integration test, not a Merge unit test :)

File: test/integration/conflict_resolution.js [hunk 5]
```diff
@@ -150,6 +150,145 @@ describe('Conflict resolution', () => {
...
   })
 
+  const afterFullSyncBothTreeDeepEqual = async (expected) => {
+    await helpers.local.scan()
+    await helpers.remote.pullChanges()
+    await helpers.syncAll()
+    await helpers.local.scan()
+    await helpers.remote.pullChanges()
+    await helpers.syncAll()
+    should(await helpers.trees()).deepEqual({remote: expected, local: expected})
```
Comment (sebn) [inline test/integration/conflict_resolution.js]: Looks like putting should.js assertions into non-test functions makes stacktraces useless... :sob:

Comment (sebn) [reply test/integration/conflict_resolution.js]: I refactored the test helpers to extract the assertions back to the test functions.

File: core/merge.js [hunk 6]
```diff
@@ -266,6 +266,9 @@ class Merge {
         log.info({path}, 'up to date')
         return null
       } else {
+        if (doc.remote._id !== folder.remote._id) {
+          await this.resolveConflictAsync(side, folder, folder)
```
Comment (taratatach) [inline core/merge.js]: Why not mark the new `doc` as the conlicting document instead?

**Comment by sebn** [comment]
I'll make separate PRs so we can merge tests, test support fix & sync fix separately.
